src/main/scala/bi/deep/SparkSchema.scala

```diff
@@ -0,0 +1,22 @@
+package bi.deep
+
+import org.apache.spark.sql.types.{StructField, StructType}
+
+class SparkSchema(
+                   var dimensions: Array[StructField],
+                   var metrics: Array[StructField]) {
+
+  lazy val mergedSchema: StructType = StructType(dimensions ++ metrics)
+  lazy val structTypeDimensions: StructType = StructType(dimensions)
+  lazy val structTypeMetrics: StructType = StructType(metrics)
```

Comment:
It is super dangerous, lazy val will be initialized once called, but your class have a `var` fields that this lazy val depends on... what happened when you update vars? This will be out of sync once called. Either use:
- `case class` with `lazy val` or
- `class` and `var` and instead of `lazy val` a `def` 
Then, there is no need for vars here, remove them